import argparse
import boto3
import mysql.connector
from mysql.connector import Error
import pandas as pd
from io import StringIO

def get_data_from_raw(endpoint_url, bucket_name, file_name="bigtech_combined.csv"):
    """Récupère les données depuis le bucket raw.
    params : 
    - file_name = bigtech_combined.csv dans le cas ou l'on exécute unpack_to_raw_v"""
    try:
        s3_client = boto3.client('s3', endpoint_url=endpoint_url)
        response = s3_client.get_object(Bucket=bucket_name, Key=file_name)
        content = response['Body'].read().decode('utf-8')
        return content
    except Exception as e:
        print(f"Erreur lors de la récupération des données depuis S3: {e}")
        return None
